jumpcloud/table_jumpcloud_user_group.go

```diff
@@ -0,0 +1,225 @@
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		// Return if no data
+		if len(userGroups) == 0 {
+			break
+		}
+
+		// Else, set the skip param to list remaining resources.
+		// The attribute skip will skip the first n resources that are already listed,
+		// and start from the immediate next to list the remaining resources.
+		localVarOptionals["skip"] = int32(resourceCount)
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getJumpCloudUserGroup(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Create client
+	client, err := getV2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("jumpcloud_user_group.getJumpCloudUserGroup", "connection_error", err)
+		return nil, err
+	}
+
+	var userGroupID string
+	if h.Item != nil {
+		userGroupID = h.Item.(v2.UserGroup).Id
+	} else {
+		userGroupID = d.EqualsQualString("id")
+	}
+
+	// Required quals cannot be empty
+	if userGroupID == "" {
+		return nil, nil
+	}
+
+	data, resp, err := client.UserGroupsApi.GroupsUserGet(ctx, userGroupID, "application/json", "application/json", nil)
+	if err != nil {
+		plugin.Logger(ctx).Error("jumpcloud_user_group.getJumpCloudUserGroup", "query_error", err)
+
+		// Ignore if resource not found error
+		if resp.StatusCode == 404 {
+			return nil, nil
+		}
+
+		// Else return the error
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func getJumpCloudUserGroupMemberships(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	var userGroupID string
+	if h.Item != nil {
+		userGroupID = h.Item.(v2.UserGroup).Id
+	} else {
+		userGroupID = d.EqualsQualString("id")
+	}
+
+	// Create client
+	client, err := getV2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("jumpcloud_user_group.getJumpCloudUserGroupMemberships", "connection_error", err)
+		return nil, err
+	}
+
+	var members []v2.GraphObject
+
+	localVarOptionals := map[string]interface{}{}
+
+	// Count the number of resource returned by the API.
+	// Set the value to 0.
+	resourceCount := 0
+
+	for {
+		data, _, err := client.UserGroupMembersMembershipApi.GraphUserGroupMembersList(ctx, userGroupID, "application/json", "application/json", localVarOptionals)
+		if err != nil {
+			plugin.Logger(ctx).Error("jumpcloud_user_group.getJumpCloudUserGroupMemberships", "query_error", err)
+			return nil, err
+		}
+
+		for _, i := range data {
+			// Increase the resource count by 1
+			resourceCount++
+
+			// append associated user details
+			members = append(members, *i.To)
+		}
+
+		// Return if no data
+		if len(data) == 0 {
+			break
+		}
+
+		// Else, set the skip param to list remaining resources.
+		// The attribute skip will skip the first n resources that are already listed,
+		// and start from the immediate next to list the remaining resources.
+		localVarOptionals["skip"] = int32(resourceCount)
+	}
```

Comment:
Since this API call has a paginated response, should we add this as a separate table?